src/node/internal/internal_tls_wrap.ts

```
@@ -0,0 +1,644 @@
+  const cb = args[1] as VoidFunction | undefined;
+
+  // If args[0] was options, then normalize dealt with it.
+  // If args[0] is port, or args[0], args[1] is host, port, we need to
+  // find the options and merge them in, normalize's options has only
+  // the host/port/path args that it knows about, not the tls options.
+  // This means that options.host overrides a host arg.
+  if (listArgs[1] !== null && typeof listArgs[1] === 'object') {
+    Object.assign(options, listArgs[1]);
+  } else if (listArgs[2] !== null && typeof listArgs[2] === 'object') {
+    Object.assign(options, listArgs[2]);
+  }
+
+  return cb ? [options, cb] : [options];
+}
+
+function onConnectSecure(this: TLSSocket): void {
+  this.authorized = true;
+  this.secureConnecting = false;
+  this.emit('secureConnect');
+
+  this[kIsVerified] = true;
+  const session = this[kPendingSession];
+  this[kPendingSession] = null;
+  if (session) this.emit('session', session);
+
+  this.removeListener('end', onConnectEnd);
+}
+
+function onConnectEnd(this: TLSSocket): void {
+  // NOTE: This logic is shared with _http_client.js
+  if (!this._hadError) {
+    const options = this[kConnectOptions];
+    this._hadError = true;
+    const error = new ConnResetException(
+      'Client network socket disconnected ' +
+        'before secure TLS connection was ' +
+        'established'
+    );
+    error.path = options?.path;
+    error.host = options?.host;
+    error.port = options?.port;
+    // @ts-expect-error TS2339 Missing types
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+    error.localAddress = options?.localAddress;
+    this.destroy(error);
+  }
+}
+
+// Arguments: [port,] [host,] [options,] [cb]
+export function connect(...args: unknown[]): TLSSocket {
+  args = normalizeConnectArgs(args);
+  let options = args[0] as NormalizedConnectionOptions;
+  const cb = args[1] as VoidFunction | undefined;
+
+  options = {
+    minDHSize: 1024,
```

We don't support setting the `minDHSize` ... perhaps we shouldn't default it and should either throw or ignore it?

---

Node.js does validate minDHSize, and that's why I added a default value here.

---

Yes, but Node.js also makes use of the option which has a very real impact on the TLS handshake. We are not using it and I really don't want to give the impression we are by silently ignoring it. I would throw if the option is provided.